adapter/src/dummy.rs

```diff
@@ -63,27 +63,43 @@ impl Adapter for DummyAdapter {
         Ok(is_same)
     }
 
-    fn validate_channel(&self, _channel: &Channel) -> AdapterResult<bool> {
-        // @TODO
-        Ok(true)
+    fn validate_channel(&self, channel: &Channel) -> AdapterResult<bool> {
+        match DummyAdapter::is_channel_valid(&self.config, channel) {
+            Ok(_) => Ok(true),
+            Err(e) => Err(AdapterError::InvalidChannel(format!("{}", e))),
+        }
     }
 
-    fn session_from_token(&self, _token: &str) -> AdapterResult<String> {
-        // @TODO
-        Ok("hello".to_string())
+    fn session_from_token(&mut self, token: &str) -> AdapterResult<Session> {
+        let mut identity = "";
```

Comment:
you should use a functional style here
`let identity = self.tokens_for_auth.iter().find(|(key, val)| val == token)`